augur/io/sequences.py

```diff
@@ -1,29 +1,57 @@
 import Bio.SeqIO
+import os
 
 from augur.errors import AugurError
+from typing import Iterator, Iterable, Union
 from .file import open_file
 
 
-def read_sequences(*paths, format="fasta"):
+def read_sequence(
+    path: str,
+    format: str = "fasta",
+) -> Bio.SeqIO.SeqRecord:
+    """Read a single sequence from a path.
+
+    Automatically infers compression mode.
+
+    Parameters
+    ----------
+    path
+        Path to a sequence file.
+
+    format
+        Format of input sequences matching any of those supported by BioPython
+        (e.g., "fasta", "genbank", etc.).
+
+    Returns
+    -------
+        A single sequence record from the given path.
+    """
+    with open_file(path) as handle:
+        return Bio.SeqIO.read(handle, format)
+
+
+def read_sequences(
```

Comment:
Given that this is using `SeqIO.parse` under the hood, perhaps `parse_sequences` would be a better name? (And would provide some visual semantic difference from `read_sequence`…)

Reply:
`read_sequences` is versioned as part of the [public API](https://docs.nextstrain.org/projects/augur/en/stable/api/public/index.html) so I'm hesitant to make changes unless they're truly necessary.

Biopython's distinction between `parse` and `read` is weird to me – there's nothing in those names that indicate plurality with `parse` and singularity with `read`, and I've found myself looking back at their docs to remember which is which. If visual semantic difference is wanted, perhaps it could be `read_sequences` vs. `read_single_sequence`.

Reply:
> `read_sequences` is versioned as part of the [public API](https://docs.nextstrain.org/projects/augur/en/stable/api/public/index.html) so I'm hesitant to make changes unless they're truly necessary.

yeah, that makes sense (unfortunate sense, but still). 

> Biopython's distinction between `parse` and `read` is weird to me – 

_nod_ yeah, agreed it feels gratuitous 

> If visual semantic difference is wanted, perhaps it could be `read_sequences` vs. `read_single_sequence`.

_nod_ That plus also, why do we need to distinguish between n=1 and n>1 in this case, feels like a lot of noise to (AFAICT) avoid the iterator in the n=1 case…

Reply:
> That plus also, why do we need to distinguish between n=1 and n>1 in this case, feels like a lot of noise to (AFAICT) avoid the iterator in the n=1 case…

This is being discussed in https://github.com/nextstrain/augur/pull/1730#discussion_r1922747392